sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsReadableSpan.java

```diff
@@ -229,6 +270,59 @@ private long getEndNanoTime() {
     }
   }
 
+  /**
+   * Returns the status of the {@code Span}. If not set defaults to {@link Status#OK}.
+   *
+   * @return the status of the {@code Span}.
+   */
+  @VisibleForTesting
+  Status getStatus() {
+    synchronized (this) {
```

Comment:
It seems less ideal to me to do locking on each of the `get*` methods. How about only synchronize `toSpanData()`, and marking these methods as `@GuardedBy("this")`?

Reply:
The synchronization mechanisms in this class are already quite problematic.  I would vote for _not_ attempting to address it in this PR (which is a 3rd or 4th attempt) and instead addressing the problems with a future PR.  I'm also happy to create an issue for that if others agree.

Reply:
Why isn't the access in here protected by a `ReadWriteLock`, rather than the manual `synchronized` accesses? It seems like we should switch to the more modern synchronization primitives, rather than using the older `synchronized` approach.

Reply:
At any rate, the code gets more complicated if I try to synchronize this whole method, due to the linter not being sophisticated enough to see that the only accesses are through method calls that are synchronized.

Reply:
@jkwatson this is a good opportunity to optimize the synchronization later. You can file an issue if you want. Thanks.

Reply:
@jkwatson this is a good opportunity to optimize the synchronization later. You can file an issue if you want. Thanks.